alf/layers.py

```diff
@@ -705,6 +706,118 @@ def bias(self):
         return self._bias
 
 
+@alf.configurable
+class CausalConv1D(nn.Module):
```

Comment:
it seems that this is only dilated conv1d? why do we need to create a new layer class?

Reply:
> it seems that this is only dilated conv1d? why do we need to create a new layer class named as "CausalConv1D"? Would be better to explicitly explain the difference between a dilated conv1d and this class in the docstring.

Not really. dilation is just an optional choice. Even without dilation, it is still different from conv1d, in that it is causal. Will add more explanations in comments.

Reply:
Now added more explanations.